drivers/memory/spiflash.h

```diff
@@ -46,24 +43,13 @@ typedef struct _mp_spiflash_cache_t {
     uint32_t block; // current block stored in buf; 0xffffffff if invalid
 } mp_spiflash_cache_t;
 
-typedef struct _mp_spiflash_config_t {
-    uint32_t bus_kind;
-    union {
-        struct {
-            mp_hal_pin_obj_t cs;
-            void *data;
-            const mp_spi_proto_t *proto;
-        } u_spi;
-        struct {
-            void *data;
-            const mp_qspi_proto_t *proto;
-        } u_qspi;
-    } bus;
-    mp_spiflash_cache_t *cache; // can be NULL if cache functions not used
-} mp_spiflash_config_t;
-
 typedef struct _mp_spiflash_t {
-    const mp_spiflash_config_t *config;
```

Comment:
The reason this config struct was separated out was so that it could be put in ROM, while the `mp_spiflash_t` struct was kept small and in RAM.  What was your reason to combine them?

Reply:
I did this early-on as I was still understanding the structure, thinking it simplified things. For boards where the interfaces are hard-coded I can see how this makes sense though, probably better if I revert this change.